From aef587c0489a9949db186398452e603fbf0433f8 Mon Sep 17 00:00:00 2001
From: Robert Joseph Evans <bobby@apache.org>
Date: Fri, 11 Jan 2013 19:26:29 +0000
Subject: [PATCH 1249/1357] MR1: MAPREDUCE-4933. MR1 final merge asks for length of file it just wrote before flushing it. Contributed by Sandy Ryza

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1432243 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 82dafad81421ef839d578b6a6da00841bfae7f34)

Reason: So that MAPREDUCE-2264 won't cause tests to fail.
Author: Sandy Ryza
Ref: CDH-7179
---
 .../org/apache/hadoop/mapred/ReduceTask.java       |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapred/ReduceTask.java b/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
index 54a9cd6..0a00288 100644
--- a/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
+++ b/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
@@ -2411,10 +2411,12 @@ class ReduceTask extends Task {
           final RawKeyValueIterator rIter = Merger.merge(job, fs,
               keyClass, valueClass, memDiskSegments, numMemDiskSegments,
               tmpDir, comparator, reporter, spilledRecordsCounter, null);
-          final Writer writer = new Writer(job, fs, outputPath,
+          Writer writer = new Writer(job, fs, outputPath,
               keyClass, valueClass, codec, null);
           try {
             Merger.writeFile(rIter, writer, reporter, job);
+            writer.close();
+            writer = null;
             addToMapOutputFilesOnDisk(fs.getFileStatus(outputPath));
           } catch (Exception e) {
             if (null != outputPath) {
-- 
1.7.0.4

